recipe/session/utils.go

```diff
@@ -146,9 +146,8 @@ func validateAndNormaliseUserInput(appInfo supertokens.NormalisedAppinfo, config
 	}
 
 	if cookieSameSite == cookieSameSite_NONE &&
-		!cookieSecure &&
-		!(topLevelAPIDomain == "localhost" || IsAnIPAPIDomain) &&
-		!(topLevelWebsiteDomain == "localhost" || IsAnIPWebsiteDomain) {
+		!cookieSecure && (!(topLevelAPIDomain == "localhost" || IsAnIPAPIDomain) ||
+		!(topLevelWebsiteDomain == "localhost" || IsAnIPWebsiteDomain)) {
```

Comment:
```suggestion
		!cookieSecure && !((topLevelAPIDomain == "localhost" || IsAnIPAPIDomain) &&
		(topLevelWebsiteDomain == "localhost" || IsAnIPWebsiteDomain)) {
```

Reply:
I find the above change easier to read, and also add comments explaining this line in detail